lib/util/result.ts

```diff
@@ -828,4 +858,26 @@ export class AsyncResult<T extends Val, E extends Val>
         )
     );
   }
+
+  onValue(fn: (value: T) => void): AsyncResult<T, E> {
+    return new AsyncResult(
+      this.asyncResult
+        .then((result) => result.onValue(fn))
+        .catch(
+          /* istanbul ignore next: should never happen */
```

Comment:
didn't you have a test for this already?

Reply:
Yes, but it's handled in the another code path, at the `Result` level, not `AsyncResult`. This code path isn't easily achievable, unless we construct the very artificial example for it.